Exercises/4-count-types.js

```diff
@@ -1,5 +1,16 @@
 'use strict';
 
-const countTypesInArray = null;
+/* const arr = [true, 3, '4', false, { n: 3 }, function (x) { return x + 1; },
+  { name: 'Marc' }, 'Marc', 3.11, [1, 2, 4], 5];*/
+
+const countTypesInArray = arr => {
+  const result = {};
+  for (const item of arr) {
+    const temp = typeof(item);
+    if (result[temp]) result[temp]++;
```

Comment:
Put `result[temp]` to a local variable

Reply:
`const counter = result[temp];`

Reply:
Yeah, but that counter is the primitive, so it would be passed by value, not by reference. The only place in code I can use it is in if-statement. In other places I still need to use result object. So what is the benefit of that counter?

Reply:
See examples here: https://github.com/HowProgrammingWorks/DataTypes/pull/21#discussion_r324387734

Reply:
By asking about putting that value into a new variable only for 'if' I meant exactly that. 
Why is it better if we still do the same quantity of callings of that object, but also every iteration we create a new variable, allocating memory for it

Reply:
Can't see any benefits of doing like that. Hope You will tell about this moment in video of code review.